### PR TITLE
Update to a post-boxo world.

### PR DESCRIPTION
**Pull request by MichaelMure**
Notably, this repo is an indirect dependency of github.com/ipfs/go-ipfs-http-client, which mess things quite fast as both boxo and non-boxo packages are imported.

### CODE REVIEW

**Comment by lidel** [comment]
@MichaelMure do you need this for any specific package? would like to know before merging it 
(we want to reduce reliance on iptb in general)

**Comment by MichaelMure** [comment]
> this repo is an indirect dependency of github.com/ipfs/go-ipfs-http-client

Dunno if that's still accurate with the move into kubo.

File: .github/workflows/go-test.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ "ubuntu", "windows", "macos" ]
-        go: ["1.19.x","1.20.x"]
+        go: ["1.20.x"]
```
Comment (Jorropo) [inline .github/workflows/go-test.yml:13]: Why ?
```suggestion
        go: ["1.19.x", "1.20.x"]
```
We usually support the two latest golang releases.

**Review by Jorropo — CHANGES_REQUESTED** [review]

File: .github/workflows/go-test.yml (hunk 1 above)
Comment (MichaelMure) [reply .github/workflows/go-test.yml:13]: Likely because of that CI fail: https://github.com/ipfs/iptb-plugins/actions/runs/4992996363

Comment (Jorropo) [reply .github/workflows/go-test.yml:13]: Can you try downgrading multierr ?
Kubo doesn't correctly work with `1.20` yet, we don't want to drop 0.19 and be cornered.

**Comment by lidel** [comment]
@MichaelMure iptb was removed when https://github.com/ipfs/go-ipfs-http-client got moved to https://github.com/ipfs/kubo/tree/master/client/rpc so we'd rather deprecate and archive this repo than spend time on maintenance like this.

But we could merge this one before that happens, as long you merge the 1.19 suggestion from Jorropo

**Comment by MichaelMure** [comment]
@lidel @Jorropo I've just done the Right Thing© and opened that PR when it came in my way. I don't need it especially, so feel free to do what you want with it, including closing it.

**Comment by aschmahmann** [comment]
triage: seems like there's not much of a need driving changes here at the moment as this is mostly used in kubo sharness tests which are doing alright at the moment. If anyone has a need for changes here we can reopen.

Thanks @MichaelMure for trying to help out here and let us know if this becomes an issue for you🙏

**Pull request closed without merging**